antsibull/lint_extra_docs.py

```diff
@@ -40,6 +40,8 @@ def lint_optional_conditions(content: str, path: str, collection_name: str
 
     Return a list of errors.
     '''
+
+    # Run rstcheck
```

Comment:
Is this comment really necessary? I mean... The next line literally calls rstcheck, it's hard to guess wrong what it does.

Reply:
Comments are never really necessary :D I added this comment earlier in this PR when this function consisted of two blocks. Now that it's back to the original block (which only does rstcheck), it's fine to remove it.

Reply:
Comments may be necessary but their purpose is to explain motivation of the choices made, not repeat what the code already says. Attempt to use them differently usually means that something is wrong with the code, like it needs refactoring/abstractions fixed. For example, if you feel the urge to divide some chunk of code into sections, there's a good chance that you could use those words as function names making the structure cleaner.